DuckDuckGoVPN/DuckDuckGoVPNAppDelegate.swift

```diff
@@ -429,10 +431,13 @@ final class DuckDuckGoVPNAppDelegate: NSObject, NSApplicationDelegate {
     private lazy var entitlementMonitor = NetworkProtectionEntitlementMonitor()
 
     private func setUpSubscriptionMonitoring() {
-        guard accountManager.isUserAuthenticated else { return }
+        guard subscriptionManager.isUserAuthenticated else { return }
 
-        let entitlementsCheck = {
-            await self.accountManager.hasEntitlement(forProductName: .networkProtection, cachePolicy: .reloadIgnoringLocalCacheData)
+        let entitlementsCheck: (() async -> Result<Bool, Error>) = {
+            Logger.networkProtection.log("Subscription Entitlements check...")
+            let isNetworkProtectionEnabled = await self.subscriptionManager.isFeatureActive(.networkProtection)
```

Comment:
[If we decide to alter it let's not do it in this PR but some follow up one. Just want to highlight it here]

I find this new naming unclear. This use case shows it:
- `isNetworkProtectionEnabled` - enabled in what sense? (can be understood as vpn installed or running), we want to denote that the user's account is granted access to the feature within their's subscription (previously we used term `entitlement`)
- `isFeatureActive` - same questions, what is active? (vpn running?)

Reply:
Noted thanks. I agree and I'll cha ge the names in future releases